### DISCUSSION TITLE
about unet_simple

### DISCUSSION THREAD

**Discussion by nirolada** [Nelson-Gon/cytounet#31] [Ideas]
What do you think about this:
The 'unet' function will use a parameter named 'double_layered' of type bool and default of False.
True: means we apply double layered convolution each stage
False: means we apply single layered convolution each stage

I know this is different from the implementation of 'unet_simple', its just an idea I had and I wondered if removing 1 layer of convolution each stage simplifies it enough for you.

**Comment by nirolada** [comment]
@Nelson-Gon you don't get notifications here or just didn't have time to reply?

**Comment by Nelson-Gon** [comment]
Hi, sorry for the delay in response, I intended to reply by the end of today. I do get notifications (mostly via email because I check my email a lot). 

I think the idea makes sense and would probably make for some good experiments regarding the effectiveness of single and double layered convolutions.  A quick search only reveals this [paper](https://papers.nips.cc/paper/2016/file/b73dfe25b4b8714c029b37a6ad3006fa-Paper.pdf) that seems to explore this issue.

**Comment by nirolada** [comment]
You said the results of the single convolution each stage are not pretty. I wonder why you did merge anyway.

**Reply by Nelson-Gon** [reply]
I merged because it does work and probably gives better results with more training. I trained for a few epochs just to [test](https://github.com/Nelson-Gon/cytounet/blob/master/examples/a549_sampler.ipynb) it out. I think it works better with more epochs and different parameters. Just needed to make sure it worked and was simpler for the merge.

**Reply by nirolada** [reply]
So the previous function, unet_simple, gave worse results?

**Reply by Nelson-Gon** [reply]
I do not quite remember. It just did not follow the actual UNet architecture so I suspended its development. As far as I remember it probably worked but I did not test it that much.
